addons/mrp/models/stock_move.py

```diff
@@ -283,3 +283,9 @@ def _get_upstream_documents_and_responsibles(self, visited):
     def _should_be_assigned(self):
         res = super(StockMove, self)._should_be_assigned()
         return bool(res and not (self.production_id or self.raw_material_production_id))
+
+    def _search_picking_for_assignation_domain(self):
+        res = super(StockMove, self)._search_picking_for_assignation_domain()
+        if self.created_production_id:
+            res.append(('move_lines.created_production_id', '!=', self.created_production_id.id))
```

Comment:
Hum, actually, the original condition was to remove a picking from the list was: `picking.move_lines.mapped('created_production_id') != self.created_production_id`

So shouldn't we take the opposite condition, i.e. `('move_lines.created_production_id', '=', self.created_production_id.id)`?